### ISSUE TITLE
Uploading WebVTT subtitles along with a video

### ISSUE THREAD

**Issue by codl** [tootsuite/mastodon#7169]
It'd be great to be able to upload one or more subtitle tracks along with a video, and select those subtitles in the video player.

I understand that the feature would see much less use than image descriptions (it is much more demanding to transcribe and time subtitles than to describe an image) but, given the opportunity, I would definitely use this, and it would be a great accessibility feature for people with a hearing disability, as well as hearing-abled people in environments where they can't listen to a video (a loud train, a library...) Having multiple tracks in different languages would also be nice.

**Comment by raphaelsiz** [comment]
wanted to second this, because while many of us uploading videos will simply add a transcription to the media description, I imagine I'm not the only person who prefers watching videos to reading transcriptions of them! Especially considering part of my audio processing issues that make me need subtitles is... dyslexia. So, because of my dyslexia, I'll have to read descriptions of videos instead of watching videos. Seems kinda counterintuitive lol

**Comment by Bluebie** [comment]
I'm already building closed caption vtt tracks for my blog, which i'm also publishing to ActivityPub. I'd love to be able to export the accessibility work i'm already doing to meaningfully appear in the activitypub post view of my blog entries. It would be great to see at least some spec work on how to represent it in AP vocab so custom implementers can make the data available for future UI consumption.

**Comment by bootblackcub** [comment]
Yes, the community group I work with would love to be able to add captions to their videos they post.

**Comment by candidexmedia** [comment]
Yes to this, or srt files

**Comment by qgustavor** [comment]
> I understand that the feature would see much less use than image descriptions (it is much more demanding to transcribe and time subtitles than to describe an image)

I guess it's not that hard nowadays because there are many audio transcription models that output subtitles. It could work like #19218. That said, I suggest there should be always a way to users submit subtitles, as even the best models have issues.

About which format to use, WebVTT is natively supported on the web, but how about non-web clients? SRT is simpler (there is no CSS fuzz to deal with) and ASS is more complex and a lot harder to implement (although there are libass builds for many platforms, including web). Converting between formats isn't that hard, ffmpeg and subtitle-edit can do it though CLI, I'm sure there are libraries for that too.
